substrate/frame/broker/src/traits.rs

```diff
@@ -0,0 +1,26 @@
+// This file is part of Substrate.
+
+// Copyright (C) Parity Technologies (UK) Ltd.
+// SPDX-License-Identifier: Apache-2.0
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// 	http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::Timeslice;
+
+pub trait NewTimesliceHook {
+	fn on_new_timeslice(t: Timeslice);
```

Comment:
Why not just put it into the `CoretimeInterface` trait?

Reply:
Because we want batching here. Without batching it may result in quite a few XCMs sent for every small reason.

Reply:
I mean put the function there instead of having another trait.